diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt

```diff
@@ -62,51 +73,516 @@ class ASTNodeUtilsTest {
+    }
+
+    @Test
+    fun `test findChildAfter`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        var counter = 0
+        applyToCode(code) { node ->
+            node.findChildAfter(VALUE_PARAMETER_LIST, TYPE_REFERENCE)?.let {
+                Assert.assertEquals("Int", it.text)
+                counter++
+            }
+        }
+        Assert.assertEquals(1, counter)
+    }
+
+    @Test
+    fun `test allSiblings withSelf - true`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            val setSibling: Set<ASTNode>?
+            val setParent: Set<ASTNode>?
+
+            setParent = if (node.treeParent != null) {
+                node.treeParent.getChildren(null).toSet()
+            } else
+                setOf(node)
+            setSibling = node.allSiblings(true).toSet()
+            Assert.assertEquals(setParent, setSibling)
+            Assert.assertTrue(setParent.isNotEmpty())
+        }
+    }
+
+    @Test
+    fun `test isNodeFromCompanionObject`() {
+
+        var code = """
+            class Something{
+            	companion object {
+                    val id = 1
+            	}
+            }
+        """.trimIndent()
+        var firstCounter = 0
+        applyToCode(code){node ->
+            if(node.elementType == PROPERTY) {
+                Assert.assertTrue(node.isNodeFromCompanionObject())
+                firstCounter++
+            }
+        }
+        code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        var secondCounter = 0
+        applyToCode(code){node ->
+            if(node.elementType == FUN) {
+                Assert.assertFalse(node.isNodeFromCompanionObject())
+                secondCounter++
+            }
+        }
+        Assert.assertEquals(1, firstCounter)
+        Assert.assertEquals(1, secondCounter)
+    }
+
+    @Test
+    fun `test isNodeFromFileLevel`() {
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        var counter = 0
+        applyToCode(code){node ->
+            if(node.treeParent != null && node.elementType == CLASS) {
+                Assert.assertTrue(node.isNodeFromFileLevel())
+                counter++
+            }
```

Comment:
Please also add one negative example - another `assertFalse` for some deeper nodes.